static/app/views/performance/newTraceDetails/traceHeader/title.tsx

```diff
@@ -0,0 +1,109 @@
+import {Fragment, useMemo} from 'react';
+import styled from '@emotion/styled';
+
+import {CopyToClipboardButton} from 'sentry/components/copyToClipboardButton';
+import ExternalLink from 'sentry/components/links/externalLink';
+import {Tooltip} from 'sentry/components/tooltip';
+import {t, tct} from 'sentry/locale';
+
+import {isRootTransaction} from '../../traceDetails/utils';
+import {isTraceNode} from '../traceGuards';
+import type {TraceTree} from '../traceModels/traceTree';
+
+const CANDIDATE_TRACE_TITLE_OPS = ['pageload', 'navigation'];
+
+interface TitleProps {
+  traceSlug: string;
+  tree: TraceTree;
+}
+
+export function Title({traceSlug, tree}: TitleProps) {
+  const traceTitle: {
+    op: string;
+    transaction?: string;
+  } | null = useMemo(() => {
+    const trace = tree.root.children[0];
+
+    if (!trace) {
+      return null;
+    }
+
+    if (!isTraceNode(trace)) {
+      throw new TypeError('Not trace node');
+    }
+
+    let firstRootTransaction: TraceTree.Title = null;
+    let candidateTransaction: TraceTree.Title = null;
+    let firstTransaction: TraceTree.Title = null;
+
+    for (const transaction of trace.value.transactions || []) {
+      const title = {
+        op: transaction['transaction.op'],
+        transaction: transaction.transaction,
+      };
+
+      if (!firstRootTransaction && isRootTransaction(transaction)) {
+        firstRootTransaction = title;
+        break;
+      } else if (
+        !candidateTransaction &&
+        CANDIDATE_TRACE_TITLE_OPS.includes(transaction['transaction.op'])
+      ) {
+        candidateTransaction = title;
+        continue;
+      } else if (!firstTransaction) {
+        firstTransaction = title;
+      }
+    }
+
+    return firstRootTransaction ?? candidateTransaction ?? firstTransaction;
+  }, [tree.root.children]);
```

Comment:
Should you be using the trace shape here to figure out what to use?

Reply:
I would also add some comments here just to understand what you are trying to do

Reply:
This replicates the logic used by the explorer endpoint to assign a name to the trace. Here's a [link](https://github.com/getsentry/sentry/blob/master/src/sentry/api/endpoints/organization_traces.py#L857-L895) to it.

Reply:
I'll add some comments too